packages/compass-schema/src/stores/store.js

```diff
@@ -217,6 +222,65 @@ const configureStore = (options = {}) => {
       }
     },
 
+    _calculateDepthByPath(input, paths) {
+      if (!input) {
+        return paths;
+      }
+      input.forEach(({types, fields, path, bsonType}) => {
+        /*
+         * Given the strucutre of schema, in case of an array,
+         * the path to an array item is still the same because indexes are not
+         * counted and for that reason we have an increment of 1 for arrays.
+         */
+        const increment = bsonType === 'Array' ? 1 : 0;
+        const score = path.split('.').length + increment;
+        if (!paths[path] || paths[path] < score) {
+          paths[path] = score;
+        }
+        this._calculateDepthByPath(types ?? fields, paths);
+      });
+
+      return paths;
+    },
+
+    calculateSchemaDepth(schema) {
+      const response = this._calculateDepthByPath(schema.fields, {});
+      const values = Object.values(response);
+      return Math.max(...values, 0);
+    },
+
+    _containsGeoData(input) {
+      let result = false;
+      if (!input) {
+        return result;
+      }
+      for (const { path, values, types, fields } of input) {
+        if (path.endsWith('.type') && intersection(MONGODB_GEO_TYPES, values).length > 0) {
+          result = true;
+        }
+        if (!result) {
+          result = this._containsGeoData(types ?? fields);
+        }
+      }
+      return result;
+    },
+
+    schemaContainsGeoData(schema) {
+      return this._containsGeoData(schema.fields);
+    },
+
+    _trackSchemaAnalyzed(analysisTime) {
+      const { schema } = this.state;
+      const trackEvent = {
+        with_filter: Object.entries(this.query.filter).length !== 0,
```

Comment:
Extreme nitpick: Above somewhere you also had a with_filter parameter, but there you did > 0 and not !== 0.